pkg/controller/registry/reconciler/grpc.go

```diff
@@ -347,31 +344,12 @@ func isRegistryServiceStatusValid(source *grpcCatalogSourceDecorator) (bool, err
 
 func (c *GrpcRegistryReconciler) ensurePod(logger *logrus.Entry, source grpcCatalogSourceDecorator, serviceAccount *corev1.ServiceAccount, defaultPodSecurityConfig v1alpha1.SecurityConfig, overwrite bool) error {
 	// currentPods refers to the current pod instances of the catalog source
-	currentPods := c.currentPods(logger, source)
-
-	var forceDeleteErrs []error
-	currentPods = slices.DeleteFunc(currentPods, func(pod *corev1.Pod) bool {
-		if !isPodDead(pod) {
-			logger.WithFields(logrus.Fields{"pod.namespace": source.GetNamespace(), "pod.name": pod.GetName()}).Debug("pod is alive")
-			return false
-		}
-		logger.WithFields(logrus.Fields{"pod.namespace": source.GetNamespace(), "pod.name": pod.GetName()}).Info("force deleting dead pod")
-		if err := c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).Delete(context.TODO(), pod.GetName(), metav1.DeleteOptions{
-			GracePeriodSeconds: ptr.To[int64](0),
-		}); err != nil && !apierrors.IsNotFound(err) {
-			forceDeleteErrs = append(forceDeleteErrs, pkgerrors.Wrapf(err, "error deleting old pod: %s", pod.GetName()))
-		}
-		return true
-	})
-	if len(forceDeleteErrs) > 0 {
-		return errors.Join(forceDeleteErrs...)
-	}
-
-	if len(currentPods) > 0 {
+	currentLivePods := c.currentPods(logger, source)
```

Comment:
Nit: Should this still be called simply `currentPods`? Including "live" in the name makes it seem like we've checked for dead-ness and excluded them in the logic of `c.currentPods()`, but that's not what's happening.

I realize it is likely they are all alive at this point because we should have already deleted the dead ones, but that happens in a separate code path and might not always be the case.

Reply:
Sure, I was just signaling that this is being reverted back to what it was originally, but I'm happy to keep it as `currentPods`.

Reply:
Done.